pkgs/servers/draupnir/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, nixosTests
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, pkgs
+, mkYarnPackage
+, fetchYarnDeps
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+
+mkYarnPackage rec {
+  pname = "draupnir";
+  version = "1.81.1";
+  # NOTE: Hard fork of Mjolnir.
+  src = fetchFromGitHub {
+    owner = "Gnuxie";
+    repo = "Draupnir";
+    rev = "v${version}";
+    sha256 = "RyVTdGnt4LbOyqYgSxbNpCYIFeYxgvlkDlFWN1Jw4DY=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    makeWrapper
+    pkgs.yarn
+    pkgs.matrix-sdk-crypto-nodejs
+  ];
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/yarn.lock";
```

Comment:
you can't do this since the yarnLock file needs to be read so that nix can evaluate the derivation. Nix can read files from derivations ("allow-from-derivation"), but using this feature is not allowed in nixpkgs. So you need to copy the lockfile.